app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -864,6 +864,28 @@ class MainViewModel @Inject constructor(
 
     private fun getActionThreads(threadsUids: List<String>): List<Thread> = threadsUids.mapNotNull(threadController::getThread)
 
+    fun getActionFolderRole(thread: Thread?): FolderRole? {
+        return getActionFolderRole(thread?.let(::listOf))
+    }
+
+    fun getActionFolderRole(message: Message): FolderRole? {
+        return getActionFolderRole(message.threads, message)
+    }
+
+    private fun getActionFolderRole(threads: List<Thread>?, message: Message? = null): FolderRole? {
```

Comment:
Add comment to precise that you can only pass list of threads with the same parent folder